### PR TITLE
RDISCROWD-7364 custom task export fields

### PR DESCRIPTION
**Pull request by n00rsy**
*Issue number of the reported bug or feature request: [RDISCROWD-7364](https://jira.prod.bloomberg.com/browse/RDISCROWD-7364)*

**Describe your changes**
 - Update task csv export: when `display_info_columns` is in the request body, filter the `task__info` fields to only contain fields in `display_info_columns`.

**Testing performed**
tested locally

### CODE REVIEW

File: pybossa/exporter/export_helpers.py [hunk 1]
```diff
@@ -213,3 +213,18 @@ def browse_tasks_export_count(obj, project_id, expanded, filters):
...
+            for header in headers:
+                if header == 'task__info':
+                    pass
+                elif header.startswith('task__info__'):
+                    masked_header = header[12:  ]
+                    if masked_header in accepted_task_info_fields_set:
+                        new_headers.append(header)
+                else:
+                    new_headers.append(header)
+            return new_headers
```
Comment (dchhabda) [inline pybossa/exporter/export_helpers.py:230]: See if following changes makes sense considering comments mentioned above
```python
def filter_table_headers(headers, filtered_columns):
  accepted_task_info_fields = set(filtered_columns) if filtered_columns else set()
  task_info_header = "task__info__"
  task_info_header_len = len(task_info_header)

  for header in headers:
    if header != 'task__info':
      if header.startswith(task_info_header):
        masked_header = header[task_info_header_len:]  
        if masked_header in accepted_task_info_fields:  
            new_headers.append(header)  
      else:  
        new_headers.append(header) 
```

File: pybossa/exporter/export_helpers.py [hunk 2]
```diff
@@ -213,3 +213,18 @@ def browse_tasks_export_count(obj, project_id, expanded, filters):
...
     return session.execute(
             sql, dict(project_id=project_id, **filter_params)).scalar()
+
+
+def filter_table_headers(headers, accepted_task_info_fields):
+            new_headers = []
+            accepted_task_info_fields_set = set(accepted_task_info_fields)
+            for header in headers:
+                if header == 'task__info':
+                    pass
```
Comment (dchhabda) [inline pybossa/exporter/export_helpers.py:223]: pass can be avoided w/ flipping the condition

Comment (n00rsy) [reply pybossa/exporter/export_helpers.py:223]: if we switch to 
```
if header != 'task__info':
 ...
else:  
     new_headers.append(header) 
```
`task__info` will always get added to `new_headers`.

Comment (n00rsy) [reply pybossa/exporter/export_helpers.py:223]: For clarity, there are 3 cases for each header:
1. header = 'task__info', so we skip it
2. header is like 'task__info__[fieldname], so we check fieldname against accepted fields set and add accordingly
3. header is something else, so we add it to the new_headers list (task__id, task__state, etc)

Comment (dchhabda) [reply pybossa/exporter/export_helpers.py]: Likely due to small tab space, it was not clear that else part of `if header != 'task__info':` won't be adding `new_header` always.

**Review by dchhabda — APPROVED** [review]
Upon making [this](https://github.com/bloomberg/pybossa/pull/988#discussion_r1823305067) changes, PR can be merged. Thanks!

**Pull request merged**